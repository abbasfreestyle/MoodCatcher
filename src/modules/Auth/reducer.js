import actionTypes from './actionTypes';

const initialState = {
  signedIn: false
};

export default (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.SIGNED_IN:
      return { ...state, signedIn: true };
    case actionTypes.SIGNED_OUT:
      return { ...state, signedIn: false };
    default:
      return state;
  }
};
