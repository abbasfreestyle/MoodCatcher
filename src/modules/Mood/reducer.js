import actionTypes from './actionTypes';

const initialState = {
  mood: 4,
  feelings: [],
  comment: null
};

export default (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.SET_MOOD:
      return { ...state, mood: action.payload.mood };
    case actionTypes.ADD_FEELING:
      return {
        ...state,
        feelings: [...state.feelings, action.payload.feeling]
      };
    case actionTypes.REMOVE_FEELING: {
      const feelings = state.feelings.filter(
        feeling => feeling.id !== action.payload.id
      );
      return { ...state, feelings };
    }
    case actionTypes.UPDATE_COMMENT:
      return { ...state, comment: action.payload.comment };
    case actionTypes.RESET_EVERYTHING:
      return initialState;
    default:
      return state;
  }
};
