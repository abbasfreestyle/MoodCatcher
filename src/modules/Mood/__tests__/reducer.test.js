import reducer, { initialState } from '../reducer';
import types from '../actionTypes';

describe('Modules > Mood > Reducer', () => {
  describe('Snapshot', () => {
    test('should match the initial state', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(initialState);
    });

    it('should handle SET_MOOD', () => {
      const action = { type: types.SET_MOOD, payload: { mood: 8 } };

      expect(reducer(initialState, action)).toEqual({
        ...initialState,
        mood: action.payload.mood
      });
    });

    it('should handle ADD_FEELING', () => {
      const action = {
        type: types.ADD_FEELING,
        payload: { feeling: { id: 1, name: 'Relaxed' } }
      };
      const addedFeeling = {
        ...initialState,
        feelings: [action.payload.feeling]
      };
      expect(reducer(initialState, action)).toEqual(addedFeeling);

      const secondAction = {
        type: types.ADD_FEELING,
        payload: { feeling: { id: 2, name: 'Happy' } }
      };
      expect(reducer(addedFeeling, secondAction)).toEqual({
        ...initialState,
        feelings: [...addedFeeling.feelings, secondAction.payload.feeling]
      });
    });

    it('should handle REMOVE_FEELING', () => {
      const action = {
        type: types.REMOVE_FEELING,
        payload: { id: 1 }
      };
      const addedFeeling = {
        ...initialState,
        feelings: [{ id: 1, name: 'Relaxed' }]
      };
      expect(reducer(addedFeeling, action)).toEqual({
        ...initialState,
        feelings: []
      });
    });

    it('should handle UPDATE_COMMENT', () => {
      const action = {
        type: types.UPDATE_COMMENT,
        payload: { comment: 'New comment' }
      };
      expect(reducer(initialState, action)).toEqual({
        ...initialState,
        comment: action.payload.comment
      });
    });

    it('should handle RESET_EVERYTHING', () => {
      const action = { type: types.RESET_EVERYTHING };
      const activeState = {
        mood: 7,
        feelings: [{ id: 1, name: 'Relaxed' }, { id: 2, name: 'Happy' }],
        comment: 'Added text'
      };
      expect(reducer(activeState, action)).toEqual(initialState);
    });
  });
});
