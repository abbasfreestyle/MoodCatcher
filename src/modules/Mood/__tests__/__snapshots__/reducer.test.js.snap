// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modules > Mood > Reducer Snapshot should match the initial state 1`] = `
Object {
  "comment": null,
  "feelings": Array [],
  "mood": 4,
}
`;
